frontend/src/metabase-lib/native.ts

```diff
@@ -4,6 +4,53 @@ import type { DatabaseId } from "metabase-types/api";
 
 import type { MetadataProvider, Query } from "./types";
 
+export function nativeQuery(
+  databaseId: DatabaseId,
+  metadata: MetadataProvider,
+  innerQuery: string,
+): Query {
+  return ML.native_query(databaseId, metadata, innerQuery);
+}
+
+export function rawNativeQuery(query: Query): string {
+  return ML.raw_native_query(query);
+}
+
+export function withNativeQuery(query: Query, innerQuery: string): Query {
+  return ML.with_native_query(query, innerQuery);
+}
+
+export function withTemplateTags(query: Query, tags: string[]): Query {
+  return ML.with_template_tags(query, tags);
+}
+
+export function templateTags(query: Query): string {
+  return ML.template_tags(query);
+}
+
+export function extractTemplateTags(
+  queryText: string,
+  existingTags?: string,
```

Comment:
existingTags - `TemplateTag` (array or an object with them?)

Reply:
Checked - it should be `Record<string, TemplateTag>`

Reply:
We have the equivalent type for this already 🎉 
https://github.com/metabase/metabase/blob/master/frontend/src/metabase-types/api/dataset.ts#L145